src/basic/utf8.c

```diff
@@ -613,6 +613,8 @@ size_t utf8_console_width(const char *str) {
 size_t utf8_last_length(const char *s, size_t n) {
         int r;
 
+        assert(s);
```

Comment:
seems unnecessary... i see nothing wrong with allowing utf_last_length(NULL, 0) to be called...

should probably be: 
```c
assert(s || n == 0);
```